torch_xla/core/xla_model.py

```diff
@@ -72,7 +72,7 @@ def is_xla_tensor(tensor):
 
 
 def parse_xla_device(device):
-  m = re.match(r'(CPU|TPU|GPU|ROCM|CUDA|XPU|NEURON):(\d+)$', device)
+  m = re.match(r'([A-Z]+):(\d+)$', device)
```

Comment:
trying to think.. where would it fail if user has a typo on `PJRT_DEVICE`?

Reply:
The device name here comes from the `PjRtClient::platform_name`, which is not guaranteed to match the `PJRT_DEVICE` name.

If there's a typo in `PJRT_DEVICE`, the runtime will fail to initialize and throw an error.